enterprise/server/githubapp/githubapp.go

```diff
@@ -1342,3 +1344,142 @@ func (a *GitHubApp) CreateGithubRef(ctx context.Context, req *ghpb.CreateGithubR
 
 	return &ghpb.CreateGithubRefResponse{}, nil
 }
+
+func (a *GitHubApp) GetGithubPullRequest(ctx context.Context, req *ghpb.GetGithubPullRequestRequest) (*ghpb.GetGithubPullRequestResponse, error) {
+	client, err := a.getGithubClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	incoming, outgoing, user, err := a.getIncomingAndOutgoingPRs(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	allIssues := append(outgoing.Issues, incoming.Issues...)
+	prs, err := a.populatePRMetadata(ctx, client, allIssues, user)
+	if err != nil {
+		return nil, err
+	}
+	resp := &ghpb.GetGithubPullRequestResponse{
+		Outgoing: []*ghpb.PullRequest{},
+		Incoming: []*ghpb.PullRequest{},
+	}
+	for _, i := range outgoing.Issues {
+		resp.Outgoing = append(resp.Outgoing, prs[i.GetNumber()])
+	}
+	for _, i := range incoming.Issues {
+		resp.Incoming = append(resp.Incoming, prs[i.GetNumber()])
+	}
```

Comment:
you can cheat since append will take a varargs, and do:
```
resp.Outgoing = append(resp.Outgoing, prs...)
resp.Incoming = append(resp.Incoming, prs...)
```

Reply:
You already do this yourself on L1357 :P

Reply:
It's slightly different because I'm looking up the incoming pr in the `prs` map (which is a `ghpb.PullRequest` that contains metadata rather than a `github.IssuesSearchResult` which is what `outgoing.Issues` is a list of) using the pull request number and appending that.

This comment did help me realize though that PRs across repos can have the same PR number, so I switched this to github's global [NodeId](https://docs.github.com/en/graphql/guides/using-global-node-ids) instead.